src/backtrace.rs

```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: 2023 Rivos Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+/// Get an array of backtrace addresses.
+///
+/// This needs `force-frame-pointers` enabled.
+use core::arch::asm;
+
+pub const MAX_BACKTRACE_ADDRESSES: usize = 10;
+
+extern "C" {
+    static _stack_start: u8;
+    static _stack_end: u8;
+}
+
+pub(crate) fn backtrace() -> [Option<usize>; MAX_BACKTRACE_ADDRESSES] {
+    // Safe because we are just reading a register
+    let fp = unsafe {
+        let mut _tmp: u64;
```

Comment:
Nit: _ should be used only for unused variables (unless it's being used to pacify the compiler in this context).

Reply:
fixed